web/src/compose_validate.ts

```diff
@@ -795,3 +798,82 @@ export function convert_mentions_to_silent_in_direct_messages(
     const silent_mention_text = people.get_mention_syntax(full_name, user_id, true);
     return silent_mention_text;
 }
+
+export function is_split_messages_enabled(): boolean {
+    return split_messages_enabled;
+}
+
+export function set_split_messages_enabled(enable: boolean): void {
+    split_messages_enabled = enable;
+}
+
+/**
+ *
+ * @param content
+ * @returns index of the delimiter occurring outside code blocks since no splitting should
+ * be done inside code blocks
+ */
+function delimiter_index_outside_code_block(content: string): number {
+    let index = content.indexOf(SPLIT_DELIMITER);
+    // skip delimiters which occur inside code blocks
+    while (index !== -1) {
+        if (!compose_ui.position_inside_code_block(content, index + 1)) {
+            return index;
+        }
+        // find next occurrence of the delimiter
+        index = content.indexOf(SPLIT_DELIMITER, index + 1);
+    }
+    return -1;
+}
+
+/**
+ * Splits a message into 2 parts, if possible
+ * @param message_content the content to be split
+ * @returns a pair of strings: the first being the individual split message, the second being the
+ * remaining part of the message, which will be empty when there are no further parts to send
+ */
+export function split_message(message_content: string): [string, string] {
+    if (!is_split_messages_enabled()) {
+        return [message_content, ""];
+    }
+    const index = delimiter_index_outside_code_block(message_content); // first occurrence of a valid delimiter
```

Comment:
We don't use inline comments... Put it on its own line above the relevant line of code